pkg/api/timestamp.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright The Rekor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package api
+
+import (
+	"context"
+	"encoding/asn1"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/go-openapi/runtime/middleware"
+	"github.com/pkg/errors"
+	"github.com/sigstore/rekor/pkg/generated/restapi/operations/timestamp"
+	"github.com/sigstore/rekor/pkg/log"
+	"github.com/sigstore/rekor/pkg/util"
+)
+
+func RequestFromRekor(ctx context.Context, requestBytes []byte) ([]byte, error) {
+	req, err := util.ParseTimestampRequest(requestBytes)
```

Comment:
perhaps we should bump this out into the calling method and return HTTP 400 (Bad Request) with this error message if we can't parse the input?